### PR TITLE
hng64 - add flat shaded polygons, with no texture, no lighting

### PR DESCRIPTION
**Pull request by Happy-yappH**

### CODE REVIEW

File: src/mame/video/hng64_3d.hxx [hunk 1]
```diff
@@ -612,6 +618,9 @@ void hng64_state::recoverPolygonBlock(const uint16_t* packet, int& numPolys)
 					currentPoly.vert[m].normal[1] = uToF(chunkOffset[10 + (9*m)]);
 					currentPoly.vert[m].normal[2] = uToF(chunkOffset[11 + (9*m)]);
 					currentPoly.vert[m].normal[3] = 0.0f;
+
+					if ( currentPoly.flatShade )
```
Comment (cuavas) [inline src/mame/video/hng64_3d.hxx:622]: Please make the code consistent – don’t put extra spaces inside the parentheses like this.

File: src/mame/includes/hng64.h [hunk 2]
```diff
@@ -61,6 +63,7 @@ struct polygon
 
 	float faceNormal[4];        // Normal of the face overall - for calculating visibility and flat-shading...
 	int visible;                // Polygon visibility in scene
+	int flatShade;              // Flat shaded polygon, no texture, no lighting
```
Comment (cuavas) [inline src/mame/includes/hng64.h:66]: Is there a reason not to use `bool` for this?

**Comment by andrew-gardner** [comment]
At a high level, this makes sense to me.  Thanks for the addition, and good sleuthing about where the constant color is!

File: src/mame/includes/hng64.h (hunk 2 above)
Comment (Happy-yappH) [reply src/mame/includes/hng64.h:66]: No, but visible also seemed like it should be a bool

Comment (DavidHaywood) [reply src/mame/includes/hng64.h:66]: the code is very old, IIRC at the time it was written bools were discouraged.

Comment (MooglyGuy) [reply src/mame/includes/hng64.h:66]: Yes, this is true. Given the heavy emphasis on matching the existing code style, it's fine for it to be an int for now. However, there should be a GitHub issue filed as a to-do reminder to have a pass over HNG64 to bring it up to somewhat modern standards. I might even take a shot at it tonight. But it shouldn't be a blocking issue for merging this PR.

**Pull request merged**